docs/how-to-guides/work-with-triggers.md

```diff
@@ -0,0 +1,84 @@
+
+Memgraph supports **database triggers** that can be executed if a particular type of event occurs. Events are related to changes in data, such as created, updated, and deleted nodes or relationships. 
+
+[![Related - Reference Guide](https://img.shields.io/static/v1?label=Related&message=Reference%20Guide&color=yellow&style=for-the-badge)](/reference-guide/triggers.md) 
+
+
+## How to create a trigger?
+
+If you need a trigger after the node(vertex) has been created, you can set up the following trigger: 
+
+```cypher
+CREATE TRIGGER createVertex
+ON CREATE AFTER COMMIT EXECUTE
+UNWIND createdVertices AS createdVertex
+SET createdVertex.created = timestamp()
+```
+
+Here the trigger's name is `createdVertex`,  `ON CREATE` defines what operation will trigger execute. `AFTER COMMIT EXECUTE` means the trigger will be executed after changes have been committed to the database. For the sake of ease of use, triggers have a set o **predefined variables**. One is `createdVertices`, a list of all created nodes (vertices). To find a complete list of predefined variables, supported operations, and configuration details, look at triggers [reference guide](https://memgraph.com/docs/memgraph/reference-guide/triggers).
+
+The trigger for updates on the nodes is almost the same. Notice different predefined variable and `ON UPDATE` command:  
+
+```cypher
+CREATE TRIGGER updateVertex
+ON UPDATE AFTER COMMIT EXECUTE
+UNWIND updatedVertices AS updatedVertex
+SET updatedVertex.updated_at = timestamp()
+
+```
+
+If you want to create the trigger that will be executed on the creation of a node or relationship, you can do it by using the predefined variable `createdObjects`.  
+
+```cypher
+CREATE TRIGGER exampleTrigger
+ON CREATE AFTER COMMIT EXECUTE
+UNWIND createdObjects AS createdObject
+WITH CASE
+        WHEN createdObject.vertex IS NOT null THEN createdObject.vertex
+        WHEN createdObject.edge IS NOT null THEN createdObject.edge
+    END AS object
+SET object.created_at = timestamp();
+```
+
+Variable `createdObjects` is list of  maps that contains `edge` or `vertex` and `event_type` property. Both depend on 
+what object has been created. If your Query is creating multiple nodes and relationships, they will be in a separate created objects maps. 
+
+
+## How to create trigger for Python query module? 
+
+Triggers can be used for executing query modules. In this sample trigger will call `query_module.new_edge(edge)` on each new 
+relationship (edge) creation:
+
+```cypher
+CREATE TRIGGER newEdge
+ON CREATE BEFORE COMMIT EXECUTE
+UNWIND createdEdges AS edge
+CALL query_module.new_edge(edge) YIELD *;
+```
+
+Make sure that your function accepts proper Memgraph type, `mgp.Edge` in this case: 
+
+```python
+def new_edge(
+    context: mgp.ProcCtx,
+    edge: mgp.Edge
+)
+```
+If you are not sure where `mgp.Edge` comes from, feel free to check the reference guide on [Python API](https://memgraph.com/docs/memgraph/reference-guide/query-modules/api/python-api)
```

Comment:
I think I would explain the mgp. Edge here and then say - for more information about python objects (or whatever).
Remember about not really wanting to click away from the material you are reading at the moment... i want to know NOW what mgp. edge is, and if I'll explore stuff deeper, then I'll check the API docs

Reply:
I agree with the comment.  ut I would left it below everthing because we are not focusing on what Python API but on the triggers.